tests/test_docs/test_commands.py

```diff
@@ -0,0 +1,203 @@
+        tree = deepcopy(self.tree)
+        latest_subcmd = None
+        allow_option_arg = False
+
+        cmd_parts = [i for i in cmd.split(" ") if i]
+
+        # Iterate the command parts
+        for cmd_part in cmd_parts:
+
+            # Subcommands
+            if cmd_part in tree["commands"].keys():
+                latest_subcmd = cmd_part
+                tree = tree["commands"][cmd_part]
+                allow_option_arg = False
+                continue
+
+            # Options
+            if cmd_part.startswith("-"):
+                if cmd_part not in tree["options"]:
+                    print(
+                        f"Command validation error in {file_}: option '{cmd_part}' is not present on the command tree {list(tree['options'])}:\n    {cmd}"
+                    )
+                    return False
+
+                allow_option_arg = True
+                continue
+
+            # Option arguments: we can't validate them, just guess that they are correct.
+            if allow_option_arg:
+                allow_option_arg = False
+                continue
+
+            # Command arguments
+            if not latest_subcmd:
+                print(
+                    f"Command validation error in {file_}: detected argument '{cmd_part}' but no latest subcommand exists yet:\n    {cmd}"
+                )
+                return False
+
+            if not tree["arguments"]:
+                print(
+                    f"Command validation error in {file_}: argument '{cmd_part}' is not valid as the latest subcommand [{latest_subcmd}] does not admit arguments:\n    {cmd}"
+                )
+                return False
+
+            # If we reach here, this command part is probably an argument for a command.
+            # We can't validate it, just guess that it is correct.
+
+        return True
+
+
+def test_validate_doc_commands() -> None:
+    """Test that doc commands are valid"""
+
+    # Get the markdown files and the Makefile
+    target_files = list(Path("docs").rglob("*.md"))
+    target_files.append(Path("Makefile"))
+
+    # Get the validator
+    validator = CommandValidator(aea_cli)
+
+    COMMAND_REGEX = (
+        r"(?P<full_cmd>(?P<cli>aea) ((?!(&|'|\(|\[|\n|\.|`|\||#|</code>|=)).)*)"
+    )
+
+    skips = [
+        "aea root, run:",
+        "aea one",
+        "aea and its plugins in your Dockerfile to avoid the pypi versions being used instead of the local ones:",
+        "aea to use the IPFS registry",
+        "aea CLI tool, the private keys will be stored in plaintext",
+        'aea run",',
+        "aea interact",
+        "aea interrupted!",
+        "aea stopping",
+        "aea here as",
+        "aea with an author",
+        "aea scaffold connection/contract/protocol/skill",
+        "aea config dir task",
+        "aea instance task",
+        "aea framework",
+        "aea project path",
+        "aea validation flag",
+        "aea instance",
+        "aea exception policies",
```

Comment:
Hm, this is an indication that either your script is too generic or the statements are wrong. If we don't refer to an AEA CLI command then the docs should use `AEA` rather than `aea` imo.